pkg/util/mount/fake.go

```diff
@@ -175,7 +179,35 @@ func (f *FakeMounter) PathIsDevice(pathname string) (bool, error) {
 }
 
 func (f *FakeMounter) GetDeviceNameFromMount(mountPath, pluginDir string) (string, error) {
-	return getDeviceNameFromMount(f, mountPath, pluginDir)
+	if runtime.GOOS == "darwin" {
+		return GetDeviceNameFromMountDarwin(f, mountPath, pluginDir)
+	} else {
+		return getDeviceNameFromMount(f, mountPath, pluginDir)
+	}
+}
+
+// getDeviceNameFromMount find the device name from /proc/mounts in which
+// the mount path reference should match the given plugin directory. In case no mount path reference
+// matches, returns the volume name taken from its given mountPath
+func GetDeviceNameFromMountDarwin(mounter Interface, mountPath, pluginDir string) (string, error) {
```

Comment:
don't export this. it's also more typical to define the same function name twice in per-os files with build tags, rather than per-os functions that remain possible to miscall accidentally

Reply:
@liggitt - I want to make sure I understand your recommendation. I should create `fake_darwin.go`, and maybe other `fake_<os>.go` files (e.g. linux and unsupported) if needed.

Reply:
yes, that is preferred over GOOS + `if` forking